src/nlp/features.py

```diff
@@ -424,12 +533,12 @@ def encode_nested_example(schema, obj):
         if isinstance(obj, str):  # don't interpret a string as a list
             raise ValueError("Got a string but expected a list instead: '{}'".format(obj))
         return [encode_nested_example(schema.feature, o) for o in obj]
-
     # Object with special encoding:
     # ClassLabel will convert from string to int, TranslationVariableLanguages does some checks
     elif isinstance(schema, (ClassLabel, TranslationVariableLanguages, Value)):
         return schema.encode_example(obj)
-
+    elif "extension<arrow.py_extension_type>" == str(schema):
+        return schema.encode_example(obj)
```

Comment:
I wonder if it's possible to use `isinstance(schema, ...)` with something like `pa.PyExtensionType` ?